packages/portfolio-contract/test/network/prod-network.test.ts

```diff
@@ -0,0 +1,151 @@
+import test from 'ava';
+import { Far } from '@endo/marshal';
+
+import { makeGraphFromDefinition } from '../../tools/network/buildGraph.js';
+import PROD_NETWORK, {
+  PROD_NETWORK as NAMED_PROD,
+} from '../../tools/network/network.prod.js';
+import type { AssetPlaceRef } from '../../src/type-guards-steps.js';
+
+const brand = Far('TestBrand') as any;
+const feeBrand = Far('TestFeeBrand') as any;
+
+const toSet = <T>(iter: Iterable<T>) => new Set(iter);
+
+// Shared expectations
+const HUBS = [
+  '@agoric',
+  '@noble',
+  '@Arbitrum',
+  '@Avalanche',
+  '@Base',
+  '@Ethereum',
+  '@Optimism',
+];
+
+const EVM_HUBS = ['@Arbitrum', '@Avalanche', '@Ethereum', '@Base', '@Optimism'];
+
+const POOLS = [
+  // Aave pools
+  'Aave_Arbitrum',
+  'Aave_Avalanche',
+  'Aave_Base',
+  'Aave_Ethereum',
+  'Aave_Optimism',
+  // Compound pools
+  'Compound_Arbitrum',
+  'Compound_Base',
+  'Compound_Ethereum',
+  'Compound_Optimism',
+  // USDN pools
+  'USDN',
+  'USDNVault',
+];
+
+// Helpers
+const getGraph = () =>
+  makeGraphFromDefinition(PROD_NETWORK, {}, {}, brand, feeBrand);
+
+const hasEdge = (
+  edges: ReturnType<typeof getGraph>['edges'],
+  src: string,
+  dest: string,
+  via?: string,
+) =>
+  edges.some(
+    e =>
+      e.src === (src as any) &&
+      e.dest === (dest as any) &&
+      (via ? e.via === (via as any) : true),
+  );
+
+const isReachable = (
+  edges: ReturnType<typeof getGraph>['edges'],
+  start: string,
+  goal: string,
+) => {
+  const q = [start as any];
+  const seen = new Set<string>([start]);
+  while (q.length) {
+    const cur = q.shift() as any;
+    if (cur === goal) return true;
+    for (const e of edges) {
+      if (e.src !== cur) continue;
+      if (!seen.has(e.dest as any)) {
+        seen.add(e.dest as any);
+        q.push(e.dest as any);
+      }
+    }
+  }
+  return false;
+};
+
+// Map a pool key to its hub based on naming convention (or noble for USDN)
+const poolToHub = (pool: string): AssetPlaceRef => {
```

Comment:
`PoolPlaces` has this info:

```suggestion
const poolToHub = (pool: string): AssetPlaceRef => `@PoolPlaces[pool].chainName`;
```